tests/qos/test_tunnel_qos_remap_nvidia.py

```diff
@@ -0,0 +1,231 @@
+            pytest_assert(wait_until(
+                5, 1, 0, check_queue_counter, rand_selected_dut, [downlink_interface], queue, PKT_NUM),
+                "The queue counter for DSCP: {}, Queue: {} is not as expected: >= {}. "
+                "Please check the output of cmd 'show queue counters' in the test log.".format(dscp, queue, PKT_NUM))
+
+    with allure.step("Start the downlink test"):
+        for dscp, queue in down_link_test_data:
+            pkt, expected_pkt = _generate_packet(dualtor_meta['vlan_mac'], server_ip, '1.1.1.1', dscp)
+            # Clear queue counters before sending traffic
+            rand_selected_dut.shell('sonic-clear queuecounters')
+            # Send packets
+            logger.info("Sending normal traffic from server {} to t1 with dscp {}".format(server_ip, dscp))
+            logger.info("The expected queue for the traffic is {}".format(queue))
+            ptfadapter.dataplane.flush()
+            testutils.send(ptfadapter, server_ptf_port, pkt, PKT_NUM)
+            receive_port_index, _ = testutils.verify_packet_any_port(ptfadapter, expected_pkt, t1_ptf_ports)
+            receive_ptf_port = t1_ptf_ports[receive_port_index]
+            pytest_assert(wait_until(
+                5, 1, 0, check_queue_counter, rand_selected_dut,
+                [uplink_interfaces[receive_ptf_port]], queue, PKT_NUM),
+                "The queue counter for DSCP: {}, Queue: {} is not as expected: >= {}. "
+                "Please check the output of cmd 'show queue counters' in the test log.".format(dscp, queue, PKT_NUM))
+
+
+def test_traffic_between_servers(ptfhost, rand_selected_dut, rand_unselected_dut,  # noqa F811
+                                 toggle_all_aa_ports_to_rand_selected_tor,  # noqa F811
+                                 tbinfo, ptfadapter): # noqa F811
+    """
+    The test case is to verify on a same tor, the traffic from an active server to a standby server should be
+    forwarded by the tunnel, and enter the correct queue.
+    """
+    # For the MSFT preserved dscp, 3/4 will be remapped, others will be copied to the outer.
+    dscp_to_queue_mapping = [
+        # DSCP, Outer DSCP, Expected queue
+        (8, 8, 0),
+        (0, 0, 1),
+        (33, 33, 1),
+        (3, 2, 2),
+        (4, 6, 6),
+        (46, 46, 5),
+        (48, 48, 7)
+    ]
+    with allure.step("Prepare test parameters and switch one server on the lower tor to active"):
+        dualtor_meta = dualtor_info(ptfhost, rand_selected_dut, rand_unselected_dut, tbinfo)
+        lower_tor_mg_facts = rand_unselected_dut.get_extended_minigraph_facts(tbinfo)
+        upper_tor_mg_facts = rand_selected_dut.get_extended_minigraph_facts(tbinfo)
+        vlan_interface = list(lower_tor_mg_facts['minigraph_vlans'].keys())[0]
+        # use the last interface as the active interface on lower tor
+        lower_tor_active_interface = lower_tor_mg_facts['minigraph_vlans'][vlan_interface]['members'][-1]
```

Comment:
Can you use existing fixture 
https://github.com/sonic-net/sonic-mgmt/blob/a12ad956235da6762212066404ed4468978fc05e/tests/common/dualtor/dual_tor_common.py#L72